### PR TITLE
zh-cn: update using_shadow_dom

### PR DESCRIPTION
**Pull request by YieldRay**
### Description

Update translation for Using_shadow_DOM

### Additional details

Source: <https://github.com/mdn/content/blob/main/files/en-us/web/api/web_components/using_shadow_dom/index.md?plain=1>

### CODE REVIEW

**Comment by YieldRay** [comment]
The `EmbedLiveSample()` macro may worth addtional check...

File: files/zh-cn/web/api/web_components/using_shadow_dom/shadowdom.svg
[the remark this answers is not in the document]
Comment (YieldRay) [reply files/zh-cn/web/api/web_components/using_shadow_dom/shadowdom.svg:1]: done

File: files/zh-cn/web/api/web_components/using_shadow_dom/index.md [hunk 1]
```diff
@@ -1,236 +1,355 @@
...
 
-现在，让我们着手实现一个示例——[`<popup-info-box>`](https://github.com/mdn/web-components-examples/tree/master/popup-info-box-web-component)（也可以查看[在线示例](https://mdn.github.io/web-components-examples/popup-info-box-web-component/)），来说明 Shadow DOM 在 custom element 中的实际运用。它包含一个图片 icon 和一段文字，并将 icon 嵌入页面之中。每当 icon 获取到焦点时，文字会在一个弹框中显示，以提供更加详细的信息。首先，在 JavaScript 文件中，我们需要定义一个叫做 `PopUpInfo` 的类，它继承自 `HTMLElement`：
+{{EmbedLiveSample("从 JavaScript 中封装")}}
 
-```js
-class PopUpInfo extends HTMLElement {
-  constructor() {
-    // 必须首先调用 super 方法
-    super();
+## Element.shadowRoot 和 "mode" 选项
```
Comment (yin1999) [inline files/zh-cn/web/api/web_components/using_shadow_dom/index.md:130]:
```suggestion
## Element.shadowRoot 和“mode”选项
```

**Review by yin1999 — COMMENTED** [review]
some suggestions

File: files/zh-cn/web/api/web_components/using_shadow_dom/index.md [hunk 2]
```diff
@@ -1,236 +1,355 @@
 ---
-title: 使用 shadow DOM
+title: 使用影子 DOM
 slug: Web/API/Web_components/Using_shadow_DOM
 ---
 
 {{DefaultAPISidebar("Web Components")}}
 
-Web components 的一个重要属性是封装——可以将标记结构、样式和行为隐藏起来，并与页面上的其他代码相隔离，保证不同的部分不会混在一起，可使代码更加干净、整洁。其中，Shadow DOM 接口是关键所在，它可以将一个隐藏的、独立的 DOM 附加到一个元素上。本篇文章将会介绍 Shadow DOM 的基础使用。
+自定义元素的一个重要方面是封装，因为自定义元素从定义上来说是可重用功能的一部分：它可以被放置在任何网页中，并且期望它能够正常工作。因此，很重要的一点是，运行在页面中的代码不应该能够通过修改自定义元素的内部实现而意外地破坏它。影子 DOM（Shadow DOM）允许你将一个 DOM 树附加到一个元素上，并且使该树的内部对于在页面中运行的 JavaScript 和 CSS 是隐藏的。
```
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/api/web_components/using_shadow_dom/index.md:8]: edited

[the remark this answers is not in the document]
Comment (YieldRay) [reply files/zh-cn/web/api/web_components/using_shadow_dom/index.md:8]: 原文是：by definition, is a piece of reusable functionality

**Review by yin1999 — APPROVED** [review]
Thanks @YieldRay

**Pull request merged**